src/main/java/org/avaje/agentloader/AgentLoader.java

```diff
@@ -151,6 +175,68 @@ private static final boolean isWindows() {
     return File.separatorChar == '\\';
   }
 
+  /**
+   * This method takes the jar:file:path-to-filename.war!/WEB-INF/jar/jar-file/ offset that is included in the
+   * url classpath and extracts out a single jar containing the files in that match that url.
+   *
+   * @param path    - full url entry in the classpath
+   * @param agentName - the agent name that we are trying to match
+   * @return null if it fails or a full path to the jar file if it succeeds
+   */
+  public static String extractJar(URL path, String agentName) {
+    String fullPath = null;
+
+    String[] jarNames = path.getPath().split(":");
+
+    if (jarNames.length >= 2) {
```

Comment:
Good. In this type of case I'd probably expect a log statement along the
lines 'JarNames length < 2? This should not happen .... '.

For the caught exceptions ... if I don't do anything with the exception
(catch and ignore) I'd always add in a comment justifying why an exception
is caught and ignored as it shows to future readers of the code that I
thought about it and why the exception can be safely ignored (as opposed to
just being slack wrt exception handling).  I know people who have spend
days diagnosing a problem only to find an empty catch block - very very
bad.  So yes, in my book any empty catch blocks need a comment to justify
the ignore treatment.

Cheers, Rob.

On 2 July 2015 at 10:42, Kefeng DENG <notifications@github.com> wrote:

> In src/main/java/org/avaje/agentloader/AgentLoader.java
> <https://github.com/avaje-common/avaje-agentloader/pull/3#discussion_r33732653>
> :
>
> > @@ -151,6 +175,68 @@ private static final boolean isWindows() {
> >      return File.separatorChar == '\\';
> >    }
> >
> > +  /**
> > +   * This method takes the jar:file:path-to-filename.war!/WEB-INF/jar/jar-file/ offset that is included in the
> > +   * url classpath and extracts out a single jar containing the files in that match that url.
> > +   *
> > +   * @param path    - full url entry in the classpath
> > +   * @param agentName - the agent name that we are trying to match
> > +   * @return null if it fails or a full path to the jar file if it succeeds
> > +   */
> > +  public static String extractJar(URL path, String agentName) {
> > +    String fullPath = null;
> > +
> > +    String[] jarNames = path.getPath().split(":");
> > +
> > +    if (jarNames.length >= 2) {
>
> Yes. You are right. Either file protocol and jar protocol will return 2 or
> 3. There is no case that returns < 2. I will update it tonight.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/avaje-common/avaje-agentloader/pull/3/files#r33732653>
> .
>